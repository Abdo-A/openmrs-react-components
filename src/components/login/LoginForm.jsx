import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import FieldInput from '../form/FieldInput';
import Errors from '../errors/Errors';
import { Button, ButtonToolbar, Grid, Row, Col, Form, FormGroup, ControlLabel, Label } from 'react-bootstrap';


const LoginForm = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <div>
      <h3><Label>Login</Label></h3>
      <Errors/>
      <Form horizontal onSubmit={handleSubmit}>
        <Grid>

          <Row>
            <FormGroup controlId="formUsername">
              <Col componentClass={ControlLabel} sm={2}>
                Username
              </Col>
              <Col sm={4}>
                <Field name="username" type='text' component={FieldInput} placeholder="username"  />
              </Col>
            </FormGroup>
          </Row>

          <Row>
            <FormGroup controlId="formPassword">
              <Col componentClass={ControlLabel} sm={2}>
                Password
              </Col>
              <Col sm={4}>
                <Field name="password" type='password' component={FieldInput} placeholder="password"  />
              </Col>
            </FormGroup>
          </Row>

          <Row>
            <FormGroup controlId="formSubmit">
            <Col smOffset={2} sm={4}>
            <ButtonToolbar>
                <Button
                  bsStyle="success"
                  bsSize="large"
                  disabled={pristine || submitting}
                  type="submit"
                >
                  Submit
                </Button>


                <Button
                  bsStyle="danger"
                  bsSize="large"
                  disabled={pristine || submitting}
                  onClick={reset}
                >
                  Clear Values
                </Button>
            </ButtonToolbar>
            </Col>
            </FormGroup>
          </Row>
        </Grid>
      </Form>
    </div>
  );
};

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  reset: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};


export default reduxForm({
  form: 'login-form' // a unique identifier for this form
})(LoginForm);






