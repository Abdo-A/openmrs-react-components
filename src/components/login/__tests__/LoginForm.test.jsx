import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import LoginForm from '../LoginForm';


describe("loginForm", () => {

  const mockStore = configureMockStore();

  it("should render correctly", () => {

    const store = mockStore({});

    const rendered = renderer.create(
      <Provider store={store}>
        <LoginForm />
      </Provider>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

});
