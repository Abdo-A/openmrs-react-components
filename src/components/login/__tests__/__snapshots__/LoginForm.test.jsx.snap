// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LoginForm should render correctly 1`] = `
<form
  className="panel"
  onSubmit={[Function]}
>
  <div
    className="midPanelItemContainer"
  >
    <input
      className="form-control"
      id={undefined}
      max={undefined}
      min={undefined}
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
  </div>
  <div
    className="midPanelItemContainer"
  >
    <input
      className="form-control"
      id={undefined}
      max={undefined}
      min={undefined}
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="midPanelItemContainer"
  >
    <div>
      <select
        class="locationSelector"
        disabled={undefined}
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Location
        </option>
      </select>
    </div>
  </div>
  <div
    className="bottomPanelItemContainer"
  >
    <button
      className="loginButton btn btn-default"
      disabled={true}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;
